magma/lib/magma/loader/record_entry.rb

```diff
@@ -135,7 +135,7 @@ def record_exists?
 
     def check_document_validity
       @complaints = []
-      @loader.validator.validate(@model, @record) do |complaint|
+      @loader.validator.validate(@model, @record_name, @record) do |complaint|
```

Comment:
This seems in reverse with `validate` below.

Is the order model, document, record_name or model, record_name, document?  Seems contradicting.